src/components/browse/components/FacetList/RangeFacet.js

```diff
@@ -70,12 +66,61 @@ export function getRangeValuesFromFiltersByField(facets = [], filters = []){
 }
 
 
+/**
+ * Formats range facet value to be smaller to fit into FacetList & similar.
+ *
+ * @param {function} termTransformFxn - Schemas.Term.toName passed in from portal app.
+ * @param {{ field: string, field_type: string }} fieldFacetObj - Facet definition from backend.
+ * @param {number|string} rangeValue - Value to transform.
+ * @param {boolean} allowJSX - Passed to termTransformFxn.
+ */
+export function formatRangeVal(termTransformFxn, fieldFacetObj, rangeValue, allowJSX = true){
+    const { field, field_type } = fieldFacetObj;
+
+    if (rangeValue === null || typeof rangeValue === "undefined") {
+        return rangeValue; // Pass thru lack of value
+    }
+
+    if (field_type === "date") {
+        return <LocalizedTime timestamp={rangeValue} localize={false} formatType="date-file" />;
+    }
+
+    if (field_type === "number"){
+        rangeValue = parseFloat(rangeValue);
+    }
+
+    let valToShow = termTransformFxn(field, rangeValue, allowJSX);
+
+    // console.log("ABCD3", valToShow, field, rangeValue);
+
+    if (typeof valToShow === "number") {
+        const absVal = Math.abs(valToShow);
+        if (absVal.toString().length <= 6){
+            // Else is too long and will go thru toPrecision or toExponential.
+            if (absVal >= 1000) {
+                valToShow = decorateNumberWithCommas(valToShow);
+            } else {
+                // keep valToShow
+            }
+        } else {
+            valToShow = valToShow.toPrecision(3);
+            // Try to prevent trailing 0s e.g. in 0.00000100
+            // Taken from https://stackoverflow.com/questions/26299160/using-regex-how-do-i-remove-the-trailing-zeros-from-a-decimal-number
+            valToShow = valToShow.replace(/(\.\d*?[1-9])0+$/g, "$1");
+        }
+    } // else is assumed to be valid JSX already
+
+    return valToShow;
```

Comment:
Looks like large decimal values are [no longer being rounded](https://gyazo.com/231252e80b4cb408079b96ce2831fee1). I remember going back and forth with Burak about this and he was pretty adamant about keeping those dropdown values rounded similarly to the items in the value indicator (around 3/4 significant digits). If there's a better/more consistent way than the `toExponential` method I was using (did notice some occasional weirdness with that: [from prod](https://gyazo.com/c7763575a6dff1228cd7bdaa616cd0f9)), let's do that (maybe we can just do `toPrecision(3 or 4)`, idk). 

Still not sure if this was his personal nitpick or something more general, but that was definitely something he noticed and got frustrated about every time I secretly changed it back. So maybe somewhat important for readability.